armi/reactor/blueprints/__init__.py

```diff
@@ -393,69 +394,77 @@ def _resolveNuclides(self, cs):
 
         actives = set()
         inerts = set()
-        undefBurnChainActiveNuclides = set()
         if self.nuclideFlags is None:
             self.nuclideFlags = isotopicOptions.genDefaultNucFlags()
-
+        originalFlagKeys = set(self.nuclideFlags)
         isotopicOptions.autoUpdateNuclideFlags(cs, self.nuclideFlags)
```

Comment:
There might be an opportunity for a refeactor where we auto-update after the nuclide expansion code below (since the auto-updated nuclides almost certainly dont need expansion) but that would be a bit bigger of a change and could have negative unforeseen consequences . happy to try that direction though if someone wants to review it.

The upside is you wouldn't need to store `originalFlagKeys` and do all that testing as you are doing the expansion. you would.

1.  expand original flags
2.  store flags in nucsToForceInXsGen
3.  call autoUpdate

Reply:
@jakehader, do you know if there is a problem moving `autoUpdateNuclideFlags` (or some tweaked version of it) to after all the ele expansion code?

Reply:
See comment below because we might be able to just remove all this new logic with a short change in the lattice physics writer. I could be wrong though.

Reply:
we cannot. see comment below. do you have any thoughts on the above question?